### PR TITLE
DM-25824: Rotator/CCW synchronization test

### PR DESCRIPTION
**Pull request by tribeiro**

### CODE REVIEW

File: examples/Test CSC comm.ipynb [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+   "source": [
+    "# Introduction\n",
+    "\n",
+    "This notebooks show how to finish the setup and test the communication with a CSC.\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "The first step, if you are planning on running this notebook is to copy it to your home folder. This will prevent multiple users from running the same notebook.\n",
+    "\n",
+    "Next, before you can proceed to communicate with CSCs, make sure you loadded the correct notebook environment. In the upper right-hand side of the notebook there must be a `LSST` written before the empty circle, just bellow the notebook tabs. \n",
```
Comment (r-owen) [inline examples/Test CSC comm.ipynb]: bellow -> below

Consider changing "there must be a `LSST` written before the empty circle" to something like `you must see `LSST` before the empty circle`

loadded -> loaded

(Why does copying this notebook to the home folder make any difference in having multiple users running it? I am ignorant as to how the notebook server environment works.)

File: examples/Test CSC comm.ipynb [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    "Once you've done that you are ready to go. Restart the notebook kernel (go to Kernel > Restart kernel) and you are all set. \n",
+    "\n",
+    "Contact Tiago Ribeiro for help: `tribeiro` on slack or tribeiro@lsst.org. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The library we use to communicat with CSCs is called salobj. You can find a developer guide for salobj at https://ts-salobj.lsst.io. A user guide will be made available soon. \n",
```
Comment (r-owen) [inline examples/Test CSC comm.ipynb]: communicat -> communicate

File: examples/Test CSC comm.ipynb [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    "\n",
+    "```\n",
+    "source $OSPL_HOME/release.com\n",
+    "```\n",
+    "\n",
+    "to the bottom of the file. \n",
+    "\n",
+    "Once you've done that you are ready to go. Restart the notebook kernel (go to Kernel > Restart kernel) and you are all set. \n",
+    "\n",
+    "Contact Tiago Ribeiro for help: `tribeiro` on slack or tribeiro@lsst.org. "
```
Comment (r-owen) [inline examples/Test CSC comm.ipynb]: Are you sure you want to publish your email address? (And I suggest slack -> Slack).

File: examples/Test CSC comm.ipynb [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    "The next cell starts a `Remote` for the `ATMCS`. If you know the name of the CSC you want to communicate with, you can replace the name in the cell bellow. The full list of available CSCs can be found here: https://ts-xml.lsst.io"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = salobj.Remote(salobj.Domain(), \"ATMCS\")"
```
Comment (r-owen) [inline examples/Test CSC comm.ipynb]: Please save the domain so you can instantiate other Remotes with it. That makes a better example. Plus you can more easily close the domain when you are done with it. In other words:
```
domain = salobj.Domain()
r = salobj.Remote(domain, ...)
```

File: examples/Test CSC comm.ipynb [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    "await r.cmd_enable.start()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A couple things to note. \n",
+    "\n",
+    "Both commands are followed by a `.start()` method. This is the method that actually returns the `asyncio` task for that command to be executed. It must follow all commands you are trying to execute. \n",
```
Comment (r-owen) [inline examples/Test CSC comm.ipynb:167]: Please consider mentioning `set_start` for commands that have parameters.

File: examples/TestMTCS.ipynb [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Test MTCS\n",
+    "\n",
+    "This notebook contains some examples on how to use and test the `MTCS` class for high level control.\n",
+    "\n",
+    "One benefit of using the `MTCS` class instead of instantiating the remotes individually is that you get all the remotes for the components in addition to the high level tasks.\n",
+    "\n",
+    "For now the MTCS only have limited funciontality implemented and they are not well tested and sanctioned. Some generic behavior (like `enable`) also have some limitations as some components are not adhering to the architecture. For instance, M1M3 is not sending the `settingVersions` event with information about configurations but it requires a configuration to be specified (e.g. `Default`). This is worked around by providing the configuration explicitly when calling the `enable` method, as one can see bellow. \n"
```
Comment (r-owen) [inline examples/TestMTCS.ipynb]: functiontality -> functionality

bellow -> below (you might do a global search for this one)

File: examples/TestMTCS.ipynb [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+    "        pass"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# await mtcs.rem.rotator.cmd_clearError.start()"
```
Comment (r-owen) [inline examples/TestMTCS.ipynb:152]: Please consider deleting this cell, or at least adding a comment explaining why it exists

File: examples/TestMTCS.ipynb [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+    "await mtcs.rem.newmtmount.cmd_enableCameraCableWrapTracking.start()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Slew to a circumpolar target\n",
+    "\n",
+    "This next cell will slew to `HD 195875` which is a circumpolar target, hence, always visible at the observatory. One detail to keep in mind is that the rotator position might have to be tweaked. If the command fails with an exception like \"rotator position out of range\" try changing to another value."
```
Comment (r-owen) [inline examples/TestMTCS.ipynb:189]: Is it worth using AstroPy to come up with a reasonable angle?

File: examples/integration/CCW-Rotator-IT.ipynb [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+    "The `timeAndDate` topic provided by the pointing contains the Local Sidereal Time, that can be used as a good indication of `RA`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time_and_date = await mtcs.rem.mtptg.tel_timeAndDate.aget()\n",
```
Comment (r-owen) [inline examples/integration/CCW-Rotator-IT.ipynb:216]: Why not use the `azElTarget` command instead? It seems much more direct as you can command exactly the angle that you want, and you don't have to worry about the current time, nor the passing of time.

The one test I can imagine wanting RA/Dec for is finding out if the CCW can track when going through the zenith cone (by "through" I mean "as near as the telescope permits")

File: tribeiro/Rotator/basic_rotator_test.ipynb [hunk 10]
```diff
@@ -219,7 +223,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.2"
+   "version": "3.6.8"
```
Comment (r-owen) [inline tribeiro/Rotator/basic_rotator_test.ipynb:226]: Why this change? It would certainly be better to run with Python 3.7 if we can get it.

File: tribeiro/Rotator/queue.ipynb [hunk 11]
```diff
@@ -0,0 +1,80 @@
+{
```
Comment (r-owen) [inline tribeiro/Rotator/queue.ipynb:1]: What is this file for? It doesn't seem to have any comments.

File: examples/EFD_data_analysis/CCW-Rotator-EFD.ipynb [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# DM-25824: Rotator/CCW synchronization test\n",
+    "\n",
+    "This notebook contains queries to the EFD to analyse the data from the CCW-Rotator synchronization test.\n",
+    "\n",
+    "The test consists of performing repetitive movements with the Rotator and make sure that the CCW can keep up.\n",
```
Comment (r-owen) [inline examples/EFD_data_analysis/CCW-Rotator-EFD.ipynb]: Consider "and make sure that" -> "to see if"

File: examples/EFD_data_analysis/CCW-Rotator-EFD.ipynb [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+    "\n",
+    "The test can be driven by a notebook (`CCW-Rotator-IT.ipynb`) or using a Script (`maintel/integration_tests/ccw_rotator_stress.py` in `ts_standardscripts`)."
```
Comment (r-owen) [inline examples/EFD_data_analysis/CCW-Rotator-EFD.ipynb:13]: I will confess it makes me very nervous to have two different ways to do the same thing. It's hard to believe that both will be maintained.

File: examples/EFD_data_analysis/CCW-Rotator-EFD.ipynb [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+    "client = lsst_efd_client.EfdClient('ncsa_teststand_efd')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Querying data from EFD\n",
+    "\n",
+    "The time set bellow is for when the test was originally performed at NCSA. This can be modified to query/analyse data taken on different runs."
```
Comment (r-owen) [inline examples/EFD_data_analysis/CCW-Rotator-EFD.ipynb]: bellow -> below (here and elsewhere)

**Review by r-owen — APPROVED** [review]
Looks good. My main suggestion is to consider using the `azElTarget` command to command the rotator, as I think it would simplify the code.

File: examples/integration/CCW-Rotator-IT.ipynb (hunk 9 above)
Comment (tribeiro) [reply examples/integration/CCW-Rotator-IT.ipynb:216]: `azElTarget` won't make the telescope track at the end of the slew...
